app/src/main/java/com/android/streetworkapp/ui/map/Map.kt

```diff
@@ -0,0 +1,55 @@
+package com.android.streetworkapp.ui.map
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.util.Log
+import androidx.activity.result.ActivityResultLauncher
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat
+import com.android.streetworkapp.ui.theme.WoodCharcoal
+
+@Composable
+fun MapManager(requestPermissionLauncher: ActivityResultLauncher<String>) {
+  val context = LocalContext.current
+  val hasLocationPermission by remember {
+    mutableStateOf(
+        ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) ==
+            PackageManager.PERMISSION_GRANTED)
+  }
+
+  // Request location permission if not granted
+  LaunchedEffect(hasLocationPermission) {
+    if (!hasLocationPermission) {
+      Log.d("Map", "requestPermissionLauncher.launch")
```

Comment:
What happen when `hasLocationPermission` is True ?

Reply:
When `hasLocationPermission` is True the LaunchedEffect does not request the location permission since the permission is already granted